packages/web-components/src/components/ic-badge/ic-badge.tsx

```diff
@@ -0,0 +1,237 @@
+import {
+  Component,
+  Element,
+  Host,
+  Method,
+  Prop,
+  State,
+  h,
+} from "@stencil/core";
+import {
+  IcBadgePositions,
+  IcBadgeTypes,
+  IcBadgeVariants,
+} from "./ic-badge.types";
+import {
+  IcColorRGBA,
+  IcSizes,
+  IcThemeForeground,
+  IcThemeForegroundEnum,
+} from "../../utils/types";
+import {
+  getCssProperty,
+  getParentElement,
+  getParentElementType,
+  hexToRgba,
+  rgbaStrToObj,
+} from "../../utils/helpers";
+
+@Component({
+  tag: "ic-badge",
+  styleUrl: "ic-badge.css",
+  shadow: {
+    delegatesFocus: true,
+  },
+})
+export class Badge {
+  private customColorRGBA: IcColorRGBA;
+  private foregroundColour: IcThemeForeground;
+
+  @Element() el: HTMLIcBadgeElement;
+
+  @State() visible: boolean = true;
+
+  /**
+   * The accessible label of the badge component to provide context for screen reader users.
+   */
+  @Prop() accessibleLabel?: string;
```

Comment:
I'm still getting 'badge displaying undefined' on the buttons when prop this isn't provided. 

And on the tabs, it reads out the tab label now but then reads out just the number on the badge e.g. "Small 1" for the first tab